web/packages/teleport/src/Account/Account.tsx

```diff
@@ -163,13 +168,197 @@ export function Account({
+    0x00010416:'Portuguese (Brazilian ABNT2)',
+    0x00010418:'Romanian (Standard)',
+    0x00010419:'Russian (Typewriter)',
+    0x0001041b:'Slovak (QWERTY)',
+    0x0001041e:'Thai Pattachote',
+    0x0001041f:'Turkish F',
+    0x00010426:'Latvian (QWERTY)',
+    0x00010427:'Lithuanian',
+    0x0001042b:'Armenian Western',
+    0x0001042e:'Sorbian Extended',
+    0x0001042f:'Macedonian (FYROM) - Standard',
+    0x00010437:'Georgian (QWERTY)',
+    0x00010439:'Hindi Traditional',
+    0x0001043a:'Maltese 48-key',
+    0x0001043b:'Sami Extended Norway',
+    0x00010445:'Bengali - INSCRIPT (Legacy)',
+    0x0001045a:'Syriac Phonetic',
+    0x0001045b:'Sinhala - wij 9',
+    0x0001045d:'Inuktitut - Naqittaut',
+    0x00010465:'Divehi Typewriter',
+    0x0001080c:'Belgian (Comma)',
+    0x0001083b:'Finnish with Sami',
+    0x00011009:'Canadian Multilingual Standard',
+    0x00011809:'Gaelic',
+    0x00020401:'Arabic (102) AZERTY',
+    0x00020402:'Bulgarian (phonetic layout)',
+    0x00020405:'Czech Programmers',
+    0x00020408:'Greek (319)',
+    0x00020409:'United States - International',
+    0x00020418:'Romanian (Programmers)',
+    0x0002041e:'Thai Kedmanee (non-ShiftLock)',
+    0x00020422:'Ukrainian (Enhanced)',
+    0x00020427:'Lithuanian New',
+    0x00020437:'Georgian (Ergonomic)',
+    0x00020445:'Bengali - INSCRIPT',
+    0x0002083b:'Sami Extended Finland-Sweden',
+    0x00030402:'Bulgarian (phonetic layout)',
+    0x00030408:'Greek (220) Latin',
+    0x00030409:'United States-Devorak for left hand',
+    0x0003041e:'Thai Pattachote (non-ShiftLock)',
+    0x00040408:'Greek (319) Latin',
+    0x00040409:'United States-Dvorak for right hand',
+    0x00050409:'Greek Latin',
+    0x00060408:'Greek Polytonic',
+  }
+
+  let layout = ctx.preferences.keyboardLayout;
+  let value: Option<any, any> = {label: layouts[layout], value: layout};
+  let options: Option<any, any>[] = Object.keys(layouts).map(k=> { return {label: layouts[k], value: parseInt(k)}});
+
   return (
     <Relative>
       <FeatureBox>
         <FeatureHeader>
           <FeatureHeaderTitle>Account Settings</FeatureHeaderTitle>
         </FeatureHeader>
         <Flex flexDirection="column" gap={4}>
+          <Box>
+            <SingleRowBox>
+              <Header
+                  title="Keyboard layout"
+                  description="Keyboard layout used by Windows Desktop sessions"
+                  icon={<Icon.Edit/>}
+                  actions={
+                    <Box width="210px">
+                      <Select
+                          onChange={selected => {
+                            if (Array.isArray(selected)) {
+                              selected = selected[0];
+                            }
+                            console.log("value", selected.value)
+                            ctx.updatePreferences({
+                              keyboardLayout: selected.value
+                            }).then(() => value = selected);
```

Comment:
We don't need the code in `.then`, after changing the value the component will re-render, and the new value will be taken from `preferences`.